src/svg/qsvggenerator.cpp

```diff
@@ -934,6 +952,17 @@ bool QSvgPaintEngine::end()
     Q_D(QSvgPaintEngine);
 
     d->stream->setString(&d->defs);
+
+    for (auto it = clip_path_to_id.begin(); it != clip_path_to_id.end(); it++)
```

Comment:
You don't really need to use iterator here.  You can use range-based for loop instead
`
for ( auto path_and_id: clip_path_to_id)
{
     *d->stream << "<clipPath id=\"clip" << path_and_id->second << "\">" << '\n';
      *d->stream << "\t" << path_and_id->first.c_str() << " \"/> \n";
      *d->stream << "</clipPath>" << '\n';
}
`
}